### PR TITLE
feat(design): BREAKING CHANGE: Implement new typefaces in Atlantis [JOB-46209]

### PR DESCRIPTION
**Pull request by amia-real**
## Motivations

Updated Fonts used by Atlantis
### Added

- deps (Design): BREAKING CHANGE: Added optional dependency of @jobber/fonts to the design package

### Changed

- build(Design): BREAKING CHANGE: Changed build process to not bundle the fonts if the person does not have access to the fonts package 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/LightBox/LightBox.css [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .wrapper:global(.ril__outer) {
-  font-family: "Source Sans Pro", Helvetica, Arial, sans-serif;
+  font-family: "Jobber Pro", "Source Sans Pro", Helvetica, Arial, sans-serif;
```
Comment (chris-at-jobber) [inline packages/components/src/LightBox/LightBox.css]: We want to use Inter as our Source replacement - Jobber Pro is the in-Jobber Poppins replacement

Comment (amia-real) [reply packages/components/src/LightBox/LightBox.css]: Oh woops, did I switch them around?

Comment (chris-at-jobber) [reply packages/components/src/LightBox/LightBox.css]: Yep, swapped them back [here](https://github.com/GetJobber/atlantis/pull/952/commits/1c41b2eb62ba9573b9152baa8e45b5140cea612c) though so we're good

**Comment by chris-at-jobber** [comment]
**Jobber Pro enabled locally**

image


**Jobber Pro disabled locally**

image

File: packages/design/package.json [hunk 2]
```diff
@@ -26,20 +26,22 @@
...
     "ci:typescript-compile": "npm run build:ts"
   },
-  "dependencies": {
-    "classnames": "^2.2.6"
+  "optionalDependencies": {
+    "@jobber/fonts": "^1.0.0-pre-release.0"
   },
-  "devDependencies": {
-    "@types/classnames": "^2.2.10",
+  "dependencies": {
```
Comment (MichaelParadis) [inline packages/design/package.json:37]: Needed to move the postcss stuff to a dependency because the package needs to build when installed so that the fonts can be bundled "automagically"

File: packages/design/buildFoundation.js [hunk 3]
```diff
@@ -3,10 +3,41 @@
 const fs = require("fs");
 const postcss = require("postcss");
 const postcssCustomProperties = require("postcss-custom-properties");
+const postcssImport = require("postcss-import");
+// eslint-disable-next-line import/no-internal-modules
+const removeRule = require("./foundation_config/postcss-filter-rules");
 
-const foundation = fs.readFileSync("foundation.css");
+let hasJobberFonts = false;
```
Comment (MichaelParadis) [inline packages/design/buildFoundation.js:10]: Going to verify if this is needed again

File: package.json [hunk 4]
```diff
@@ -79,10 +81,24 @@
     "lint-staged": "^9.4.2",
     "lodash": "^4.17.21",
     "plop": "^2.5.3",
+    "css-to-react-native-transform": "^2.0.0",
     "postcss": "^7.0.36",
+    "postcss-calc": "7.0.2",
+    "postcss-cli": "^6.1.2",
```
Comment (MichaelParadis) [inline package.json:87]: I no longer think these changes are needed. I am going to verify and undo these changes if they aren't needed.

File: packages/components/package.json [hunk 5]
```diff
@@ -46,11 +46,19 @@
     "react-router-dom": "^5.2.0",
     "time-input-polyfill": "^1.0.7",
     "ts-xor": "^1.0.8",
-    "uuid": "^8.3.2"
+    "uuid": "^8.3.2",
```
Comment (MichaelParadis) [inline packages/components/package.json:49]: I also no longer think these are needed. Going to verify and remove if no longer needed

File: .circleci/config.yml [hunk 6]
```diff
@@ -44,7 +44,7 @@ references:
   npm_install: &npm_install
     run:
       name: Install Dependencies
-      command: npm ci
+      command: export IGNORE_OPTIONAL_DEPENDENCIES=true; npm ci --no-optional
```
Comment (MichaelParadis) [inline .circleci/config.yml:47]: Did this to prevent us from bundling @jobber/fonts when building releases in CI

**Comment by chris-at-jobber** [comment]
@MichaelParadis if it makes it easier to review, you could rip out my [button styling change](https://github.com/GetJobber/atlantis/pull/952/commits/d5278133003f4f770af56d583ef104b5abc8b886) as I think we already have that covered in a subsequent ticket and it adds more "stuff" here. I didn't realize I'd left that in!

File: packages/design/buildIconStyles.js [hunk 7]
```diff
@@ -27,6 +28,18 @@ postcss([
...
+        if (err) {
+          console.log("error", err);
+        }
+        files.forEach(file => {
+          console.log(file);
+        });
+      });
+    } catch (e) {
+      console.log("error", e);
+    }
```
Comment (rodrigoeidelvein) [inline packages/design/buildIconStyles.js:42]: Do we need to log everything here?

**Review by rodrigoeidelvein — COMMENTED** [review]
Hey! I only added one comment for now in the code, but how can I QA this? Only install Atlantis in a repo and try to use the fonts?

File: packages/design/buildIconStyles.js (hunk 7 above)
Comment (MichaelParadis) [reply packages/design/buildIconStyles.js:42]: I kept this from the previous version however I do not think we need to do the logging in the callback. I think logging the error is still valuable for debugging failures.

File: packages/docz-tools/src/gatsby-theme-docz/theme/typography.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
 export const fonts = {
-  body: `'Source Sans Pro', sans-serif`,
-  heading: `'Poppins', sans-serif`,
+  body: `'Inter', sans-serif`,
```
Comment (chris-at-jobber) [inline packages/docz-tools/src/gatsby-theme-docz/theme/typography.ts:2]: can we use the `var(--typography--fontFamily-normal)` in here for the body? I know these docz files are finicky so this is an earnest question 😄

Comment (MichaelParadis) [reply packages/docz-tools/src/gatsby-theme-docz/theme/typography.ts:2]: Let me double check that because docz is super finicky

Comment (MichaelParadis) [reply packages/docz-tools/src/gatsby-theme-docz/theme/typography.ts]: @chris-at-jobber sorry for the delay. I was able to remove it. Just waiting on a build now

**Comment by chris-at-jobber** [comment]
@MichaelParadis I'm still seeing Poppins rendered on the latest build when I turn off my fonts locally:

## Fonts turned off locally:

Docs:
image

Components:
image

## Fonts turned on locally:

Docs:
image

Components:
image

File: package.json [hunk 9]
```diff
@@ -99,9 +99,6 @@
     "typed-css-modules": "^0.7.0",
     "typescript": "^3.8.3"
   },
-  "config": {
```
Comment (MichaelParadis) [inline package.json]: Reminder don't override things you want to use :soblol:

**Review by rodrigoeidelvein — APPROVED** [review]
LGTM!

Tested the installation and the fonts in the docs! Thanks for this, Michael

**Comment by chris-at-jobber** [comment]
I am still seeing all of the old fonts on the Cloudflare preview build

image

**Comment by chris-at-jobber** [comment]
In the components I'm still seeing Poppins and Source as well:

image

**Comment by chris-at-jobber** [comment]
@rodrigoeidelvein can you add a screenshot of what you're seeing on your end?

**Comment by rodrigoeidelvein** [comment]
> @rodrigoeidelvein can you add a screenshot of what you're seeing on your end?

Oh yeah, it's still rendering Poppins. I don't know what I saw before, sorry about it

**Comment by MichaelParadis** [comment]
@chris-at-jobber @rodrigoeidelvein it got rid of the fonts when I resolved the conflicts apparently. Queued a new build

**Comment by chris-at-jobber** [comment]
It looks great now!

image

image
